docs/src/manual/variables.md

```diff
@@ -589,178 +908,82 @@ model[:x]
  y[2]
 ```
 
-## Integrality utilities
-
-Adding integrality constraints to a model such as `@constraint(model, x in MOI.ZeroOne())`
-and `@constraint(model, x in MOI.Integer())` is a common operation. Therefore,
-JuMP supports two shortcuts for adding such constraints.
-
-### Binary (ZeroOne) constraints
-
-Binary optimization variables are constrained to the set ``x \in \{0, 1\}``. (The
-`MOI.ZeroOne` set in MathOptInterface.) Binary optimization variables can be
-created in JuMP by passing `Bin` as an optional positional argument:
-```jldoctest variables_binary; setup=:(model=Model())
-julia> @variable(model, x, Bin)
-x
-```
-We can check if an optimization variable is binary by calling
-[`is_binary`](@ref) on the JuMP variable, and binary constraints can be removed
-with [`unset_binary`](@ref).
-```jldoctest variables_binary
-julia> is_binary(x)
-true
-
-julia> unset_binary(x)
-
-julia> is_binary(x)
-false
-```
-
-Binary optimization variables can also be created by setting the `binary`
-keyword to `true`.
-```jldoctest; setup=:(model=Model())
-julia> @variable(model, x, binary=true)
-x
-```
-### Integer constraints
-
-Integer optimization variables are constrained to the set ``x \in \mathbb{Z}``.
-(The `MOI.Integer` set in MathOptInterface.) Integer optimization variables can
-be created in JuMP by passing `Int` as an optional positional argument:
-```jldoctest; setup=:(model=Model())
-julia> @variable(model, x, Int)
-x
-```
-Integer optimization variables can also be created by setting the `integer`
-keyword to `true`.
-```jldoctest variables_integer; setup=:(model=Model())
-julia> @variable(model, x, integer=true)
-x
-```
-We can check if an optimization variable is integer by calling
-[`is_integer`](@ref) on the JuMP variable, and integer constraints can be
-removed with [`unset_integer`](@ref).
-```jldoctest variables_integer
-julia> is_integer(x)
-true
-
-julia> unset_integer(x)
-
-julia> is_integer(x)
-false
-```
-
-!!! tip
-    The [`relax_integrality`](@ref) function relaxes all integrality constraints
-    in the model, returning a function that can be called to undo the operation
-    later on.
-
 ## Semidefinite variables
 
-JuMP also supports modeling with semidefinite variables. A square symmetric
-matrix ``X`` is positive semidefinite if all eigenvalues are nonnegative. We can
-declare a matrix of JuMP variables to be positive semidefinite as follows:
+A square symmetric matrix ``X`` is positive semidefinite if all eigenvalues are
+nonnegative.
+
+Declare a matrix of JuMP variables to be positive semidefinite by passing `PSD`
+as an optional positional argument:
 ```jldoctest; setup=:(model=Model())
 julia> @variable(model, x[1:2, 1:2], PSD)
 2×2 LinearAlgebra.Symmetric{VariableRef, Matrix{VariableRef}}:
  x[1,1]  x[1,2]
  x[1,2]  x[2,2]
 ```
-or using the syntax for [Variables constrained on creation](@ref jump_variables_on_creation):
-```jldoctest; setup=:(model=Model())
-julia> @variable(model, x[1:2, 1:2] in PSDCone())
```

Comment:
Why removing this ? It should now be the prefered way, it's quite consistent with the new syntax `@constraint(model, X >= Y, PSDCone())`

Reply:
I've moved it to the constrained on creation section. I also don't know if it's the preferred way... `@variable(model, x[1:2, 1:2], PSD)` is shorter to type. I think the `x in Set` gets a bit confusing because it blurs variables and constraints.